### PR TITLE
Standardize rustdoc subheadings

### PR DESCRIPTION
**Pull request by jamillambert**
The subheadings in the rustdocs have been standardized according to [./CONTRIBUTING.md](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/CONTRIBUTING.md):
```rust
impl FooBar {
    /// Constructs a `FooBar` from a [`Baz`].
    ///
    /// # Errors
    ///
    /// Returns an error if `Baz` is not ...
    ///
    /// # Panics
    ///
    /// If the `Baz`, converted to a `usize`, is out of bounds.
    pub fn from_baz(baz: Baz) -> Result<Self, Error> {
        ...
    }
}
```

### CODE REVIEW

File: bitcoin/src/taproot/mod.rs [hunk 1]
```diff
@@ -429,8 +429,12 @@ impl TaprootBuilder {
         Ok(TaprootBuilder { branch: vec![Some(node)] })
     }
 
-    /// Adds a leaf script at `depth` to the builder with script version `ver`. Errors if the leaves
-    /// are not provided in DFS walk order. The depth of the root node is 0.
+    /// Adds a leaf script at `depth` to the builder with script version `ver`.
+    /// The depth of the root node is 0.
+    ///
+    /// # Errors
+    ///
```
Comment (apoelstra) [inline bitcoin/src/taproot/mod.rs:436]: In f52afae4fcfdd9ab013cb56bc297c7eef9c73baf:

trailing whitespace here

File: units/src/amount.rs [hunk 2]
```diff
@@ -1046,12 +1046,20 @@ impl Amount {
 
     /// Unchecked addition.
     ///
-    /// Computes `self + rhs`.  Panics in debug mode, wraps in release mode.
+    /// Computes `self + rhs`.
+    ///
+    /// # Panics
+    ///
+    /// Panics in debug mode, wraps in release mode.
```
Comment (apoelstra) [inline units/src/amount.rs:1055]: In 3f76015ee911a78c9e7cb65953166dad2614cda6:

Can you clarify this to say "**On overflow,** panics in debug mode and wraps in release mode.".

File: bitcoin/src/taproot/mod.rs (hunk 1 above)
Comment (jamillambert) [reply bitcoin/src/taproot/mod.rs:436]: Removed the white space below on line 454

File: units/src/amount.rs (hunk 2 above)
Comment (jamillambert) [reply units/src/amount.rs]: Clarified in both `unchecked_add()` and `unchecked_sub()`

**Review by tcharding — COMMENTED** [review]
Oh, you fixed review suggestions as a separate patch. My review highlights why that is not the way we do it here (some projects do).

If you put an additional patch then next reviewer (in this case me), ends up picking up the same mistakes as the first reviewer because I pull down the branch and review it patch by patch.

So in future, when you are fixing review suggestions please use `git rebase` and amend the patch that you are changing. Then do a `git push -f` to force push the new version. FTR reviewers that acked already will then use `git range-diff` to check what changed in the force push.

Remember to read all your own patches after rebasing to make sure changes are in the right patch :)

You might as well practice on this one.

File: bitcoin/src/taproot/mod.rs [hunk 3]
```diff
@@ -441,16 +445,24 @@ impl TaprootBuilder {
...
 
-    /// Adds a leaf script at `depth` to the builder with default script version. Errors if the
-    /// leaves are not provided in DFS walk order. The depth of the root node is 0.
+    /// Adds a leaf script at `depth` to the builder with default script version.
+    /// The depth of the root node is 0.
     ///
     /// See [`TaprootBuilder::add_leaf_with_ver`] for adding a leaf with specific version.
+    ///
+    /// # Errors
+    ///
```
[the remark this answers is not in the document]
Comment (tcharding) [reply bitcoin/src/taproot/mod.rs:454]: This is fixed in edfb18aa0bdb9208be035f3ee02fbe61a90456fe

File: bitcoin/src/taproot/mod.rs [hunk 4]
```diff
@@ -429,8 +429,12 @@ impl TaprootBuilder {
         Ok(TaprootBuilder { branch: vec![Some(node)] })
     }
 
-    /// Adds a leaf script at `depth` to the builder with script version `ver`. Errors if the leaves
-    /// are not provided in DFS walk order. The depth of the root node is 0.
+    /// Adds a leaf script at `depth` to the builder with script version `ver`.
+    /// The depth of the root node is 0.
```
[the remark this answers is not in the document]
Comment (jamillambert) [reply bitcoin/src/taproot/mod.rs]: Done

File: bitcoin/src/taproot/mod.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jamillambert) [reply bitcoin/src/taproot/mod.rs:456]: Done, and thanks for the shell function.

File: bitcoin/src/taproot/mod.rs [hunk 5]
```diff
@@ -441,16 +445,24 @@ impl TaprootBuilder {
...
+    ///
+    /// Errors if the leaves are not provided in DFS walk order.
     pub fn add_leaf(self, depth: u8, script: ScriptBuf) -> Result<Self, TaprootBuilderError> {
         self.add_leaf_with_ver(depth, script, LeafVersion::TapScript)
     }
 
-    /// Adds a hidden/omitted node at `depth` to the builder. Errors if the leaves are not provided
-    /// in DFS walk order. The depth of the root node is 0.
+    /// Adds a hidden/omitted node at `depth` to the builder.
+    /// The depth of the root node is 0.
```
[the remark this answers is not in the document]
Comment (jamillambert) [reply bitcoin/src/taproot/mod.rs:464]: done

**Comment by jamillambert** [comment]
Thanks for your advice.  I have removed the patch with the review suggestions and incorporated them in the original patches.  It does make more sense to do it this way.

**Review by apoelstra — APPROVED** [review]
ACK 11bb1ff6ff65e89d9f101ffb7d76ebc09a20d738

**Review by tcharding — APPROVED** [review]
ACK 11bb1ff6ff65e89d9f101ffb7d76ebc09a20d738

**Comment by tcharding** [comment]
Thanks bro.

**Comment by tcharding** [comment]
Hey @jamillambert you could fix all the warnings under a standard heading too if you feel like it. If you use `git grep -i warning` there are a bunch of non-standard things.

ref: https://github.com/rust-bitcoin/rust-bitcoin/pull/2795#discussion_r1613023238

**Pull request merged**